eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -28,61 +29,77 @@ parameters:
+          # figure out what's up with the create resource group above
+        - template: ../../../common/TestResources/deploy-test-resources.yml
+          parameters:
+            Location: ${{ coalesce(parameters.Location, cloudConfig.value.Location, 'westus2') }}
+            ServiceDirectory: '${{ parameters.ServiceDirectory }}'
+            SubscriptionConfiguration: ${{ cloudConfig.value.SubscriptionConfiguration }}
+            ArmTemplateParameters: ${{ cloudConfig.value.ArmTemplateParameters }}
+            # AdditionalParameters: $ {{ parameters.AdditionalTestArgs }}
 
-      - template: ../steps/build-test.yml
-        parameters:
-          BuildTargetingString: ${{ parameters.BuildTargetingString }}
-          ServiceDirectory: ${{ parameters.ServiceDirectory }}
-          CoverageArg: $(CoverageArg)
-          EnvVars:
-            AZURE_RUN_MODE: 'Live' #Record, Playback
-            ${{ insert }}: ${{ parameters.EnvVars }}
-          BeforeTestSteps: ${{ parameters.BeforeTestSteps }}
-          AfterTestSteps: ${{ parameters.AfterTestSteps }}
-          PythonVersion: $(PythonVersion)
-          OSVmImage: $(OSVmImage)
-          ToxTestEnv: "whl"
-          AdditionalTestArgs: ${{ parameters.AdditionalTestArgs }}
-          TestMarkArgument: ${{ parameters.TestMarkArgument }}
-          InjectedPackages: ${{ parameters.InjectedPackages }}
-          BuildDocs: ${{parameters.BuildDocs}}
+        - template: ../steps/build-test.yml
+          parameters:
+            BuildTargetingString: ${{ parameters.BuildTargetingString }}
+            ServiceDirectory: ${{ parameters.ServiceDirectory }}
+            CoverageArg: $(CoverageArg)
+            EnvVars:
+              AZURE_RUN_MODE: 'Live' #Record, Playback
+              ${{ insert }}: ${{ parameters.EnvVars }}
+            BeforeTestSteps: ${{ parameters.BeforeTestSteps }}
+            AfterTestSteps: ${{ parameters.AfterTestSteps }}
+            PythonVersion: $(PythonVersion)
+            OSVmImage: $(OSVmImage)
+            ToxTestEnv: "whl"
+            AdditionalTestArgs: ${{ parameters.AdditionalTestArgs }}
+            TestMarkArgument: ${{ parameters.TestMarkArgument }}
+            InjectedPackages: ${{ parameters.InjectedPackages }}
+            BuildDocs: ${{parameters.BuildDocs}}
 
-      - ${{ if eq(parameters['AllocateResourceGroup'], 'true') }}:
-        - pwsh: |
-            az group delete --resource-group $env:AZURE_RESOURCEGROUP_NAME -y --no-wait
-          displayName: Delete resource group
-          condition: ne(variables['AZURE_RESOURCEGROUP_NAME'], '')
+        - ${{ if eq(parameters['AllocateResourceGroup'], 'true') }}:
+          - pwsh: |
+              az group delete --resource-group $env:AZURE_RESOURCEGROUP_NAME -y --no-wait
+            displayName: Delete resource group
+            condition: ne(variables['AZURE_RESOURCEGROUP_NAME'], '')
 
+        - template: /eng/common/TestResources/remove-test-resources.yml
```

Comment:
Similar question for this it seems like we should do either the deploy/remove resources or the inline mechanisms that were there before. I also think we should file a follow-up issue to remove the AllocateResourceGroup stuff once we can prove we don't need it or if we can create a standard arm-template that will mimic the same behavior when using the scripts.

Reply:
Added this as well.